core/src/main/java/org/testcontainers/containers/DockerComposeContainer.java

```diff
@@ -662,3 +670,47 @@ private Logger logger() {
         return DockerLoggerFactory.getLogger(COMPOSE_EXECUTABLE);
     }
 }
+
+@Slf4j
+@UtilityClass
+class DockerComposePrecondition {
+    private static final String CONTAINER_NAME_OPTION = "container_name";
+    private static final String VERSION_KEY = "version";
+    private static final Yaml YAML = new Yaml(new Constructor(Map.class));
+
+    static void checkNotContainUnsupportedComposeOptions(List<File> composeFiles) {
+        composeFiles.forEach(file -> checkArgument(!containsUnsupportedComposeOptions(file),
+            String.format("Compose file %s contains '%s' option which is not supported by container.",
+                file,
+                CONTAINER_NAME_OPTION)));
+    }
+
+    @SuppressWarnings("unchecked")
+    private static boolean containsUnsupportedComposeOptions(File file) {
+        try {
+            return Optional.of(YAML.load(FileUtils.openInputStream(file)))
```

Comment:
please move `FileUtils.openInputStream(file)` to `try(...)`, otherwise it will remain open

Also, you don't need `Optional` here at all.